src/Sparrow/Json/LazyStringParser.cs

```diff
@@ -683,7 +685,17 @@ public static Result TryParseDateTime(byte* buffer, int len, out DateTime dt, ou
 
             if (buffer[4] != '-' || buffer[7] != '-' || buffer[10] != 'T' ||
                 buffer[13] != ':' || buffer[16] != ':' || buffer[16] != ':')
-                goto Failed;
+            {
+                if (len != 29 || buffer[3] != ',' || buffer[19] != ':' || buffer[22] != ':')
+                    goto Failed;
```

Comment:
All these checks can be done with SIMD instructions now. I wouldn't worry much about the difference and just create an issue to tackle this eventually.  Unless performance test in real world scenarios show important performance regressions.

Reply:
@redknightlois can you please create it? You know best how to formulate / describe it. 

Based on the above benchmark results (14.43 ns vs 13.32 ns per operation) , do you think this PR is acceptable for 5.4? 
The SIMD optimization could be targeted to 6.x / 7.x if not really needed in 5.4.